### ISSUE TITLE
Scrollspy beta 4 active state trigger points

### ISSUE THREAD

**Issue by DannyJoris** [twbs/bootstrap#23809]
I'm confused on some of the behaviour of Scrollspy, and at this point I'm not sure if it's a bug or something I did wrong.

When the scrollable element is not the default window, the offset to the page top still affects when the nav links get set to active. Note that the navigation isn't fixed position, the navigation and content aren't positioned on top of each other: https://codepen.io/DannyJoris/pen/ZJwdbv

Active states are accurate here because the scrollable div has the same offset top as the window: https://codepen.io/DannyJoris/pen/dzaxyG

Also in this case both the window scroll position _and_ the scrollable element's position get updated. This seems confusing as I'd only expect the scrollable element (the one with `data-spy="scroll"`) to update, not the window scroll: https://codepen.io/DannyJoris/pen/BdMXym

Clicking the navigation does always set the scrollable element to the correct position though. It's just that the active state trigger points seem inaccurate.

**Comment by Johann-S** [comment]
You can adjust the accuracy of our Scrollspy by using `data-offset` attribute see : https://getbootstrap.com/docs/4.0/components/scrollspy/#options

**Comment by DannyJoris** [comment]
Thanks, I guess it works when there's a fixed offset: https://codepen.io/DannyJoris/pen/eEXZLW

But what if the offset to the window top is variable? I feel like the offset should be calculated to the top of the element and not to the top of the window.

And what about the double scroll position jump?

**Comment by Johann-S** [comment]
If you have something which is variable you should call the `update` method see : https://getbootstrap.com/docs/4.0/components/scrollspy/#methods

Currently we just jump on each sections vertically not horizontally if it's what you mean by "double scroll position jump"

**Comment by DannyJoris** [comment]
With the double jump I mean this: https://codepen.io/DannyJoris/pen/BdMXym
Clicking the navigation item updates both the `.scrollspy-example-content` element offset _and_ the window offset. I would expect it should only update any element with `data-spy="scroll"`, not the window.

That's why I find it confusing that the window offset affects the offset in a contained (not `body`) `data-spy="scroll"` element.

**Comment by Johann-S** [comment]
Sorry but I don't see any change on the `.scrollspy-example-content` except the move of the scrollbar and they are no change on the window offset 😟

**Comment by DannyJoris** [comment]
Oh ok, maybe it's a cross browser issue. I'm on Chrome 60, macOS 10.12.6.

![ilcxejx6sd](https://user-images.githubusercontent.com/110365/30005700-b55d67ac-90be-11e7-9ebd-0fdf7486ceb7.gif)

**Comment by dhmskrn** [comment]
I'm on Chrome version 68.0.3440.84 (Official Build) (64-bit) Windows 7 Professional and still have the same problem.

**Comment by orianna-zzo** [comment]
> Oh ok, maybe it's a cross browser issue. I'm on Chrome 60, macOS 10.12.6.
> 
> ![ilcxejx6sd](https://user-images.githubusercontent.com/110365/30005700-b55d67ac-90be-11e7-9ebd-0fdf7486ceb7.gif)

I use scrollspy in my hugo theme and I came up with the same problem. I checked on the bootstrap.js and found that `offset` is float array (using getBoundingClientRect()) while `scrollTop` is an int. I fix this with add 1 to `scrollTop` in `_proto._process = function _process()`, so it would be `var scrollTop = this._getScrollTop() + this._config.offset + 1;`. It works well in my case.
